test/extended/etcd/vertical_scaling.go

```diff
@@ -186,4 +199,99 @@ var _ = g.Describe("[sig-etcd][Feature:EtcdVerticalScaling][Suite:openshift/etcd
+		err = errors.Wrap(err, "scale-down should not have happened before scale up when cluster membership is healthy")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 3: add a new master node and wait until it is in Running state.
+		// The new machine should belong to the same index as the deleted machine
+		// to satisfy the requirement of at least 1 Ready Replica per index.
+		deletedMachineIndex, err := strconv.ParseInt(deletedMachineName[strings.LastIndex(deletedMachineName, "-")+1:], 10, 32)
+		err = errors.Wrapf(err, "failed to parse index from deleted machine name: %s", deletedMachineName)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		newMachineName, err := scalingtestinglibrary.CreateNewMasterMachine(ctx, g.GinkgoT(), machineClient, deletedMachineIndex)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		framework.Logf("Created machine %q", newMachineName)
+
+		err = scalingtestinglibrary.EnsureMasterMachine(ctx, g.GinkgoT(), newMachineName, machineClient)
+		err = errors.Wrapf(err, "scale-up: timed out waiting for machine (%s) to become Running", newMachineName)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 4: wait for the machine pending deletion to have its member removed to indicate scale-down
+		framework.Logf("Waiting for etcd member %q to be removed", deletedMachineName)
+		deletedMemberName, err := scalingtestinglibrary.MachineNameToEtcdMemberName(ctx, oc.KubeClient(), machineClient, deletedMachineName)
+		err = errors.Wrapf(err, "failed to get etcd member name for deleted machine: %v", deletedMachineName)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		err = scalingtestinglibrary.EnsureMemberRemoved(g.GinkgoT(), etcdClientFactory, deletedMemberName)
+		err = errors.Wrapf(err, "scale-down: timed out waiting for member (%v) to be removed", deletedMemberName)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 5: Wait for apiserver revision rollout to stabilize
+		framework.Logf("waiting for api servers to stabilize on the same revision")
+		err = testlibraryapi.WaitForAPIServerToStabilizeOnTheSameRevision(g.GinkgoT(), oc.KubeClient().CoreV1().Pods("openshift-kube-apiserver"))
+		err = errors.Wrap(err, "scale-down: timed out waiting for APIServer pods to stabilize on the same revision")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// step 6: verify member and machine counts go back down to 3
+		framework.Logf("Waiting for etcd membership to show 3 voting members")
+		err = scalingtestinglibrary.EnsureVotingMembersCount(ctx, g.GinkgoT(), etcdClientFactory, kubeClient, 3)
+		err = errors.Wrap(err, "scale-down: timed out waiting for 3 voting members in the etcd cluster and etcd-endpoints configmap")
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		framework.Logf("Waiting for 3 ready replicas on CPMS")
+		err = scalingtestinglibrary.EnsureReadyReplicasOnCPMS(ctx, g.GinkgoT(), 3, cpmsClient, nodeClient)
+		err = errors.Wrap(err, "scale-down: timed out waiting for CPMS to show 3 ready replicas")
+		o.Expect(err).ToNot(o.HaveOccurred())
```

Comment:
The CPMS is disabled for this test right? Can you confirm if this would even get updated as we add and remove machines manually?
I suspect this is just verifying the pre-test 3 machine count. If so, we don't really need this check.

Reply:
@hasbro17 Even though CPMS is disabled, `cpms.Status.ReadyReplicas` and `cpms.Status.Replicas` still gets updated. I verified this. PFA images showing the changes in these values during vertical scaling. 
![Screenshot 2024-10-03 at 16 25 25](https://github.com/user-attachments/assets/ae4b7207-5627-4c5a-abce-0196bef7f480)
![Screenshot 2024-10-03 at 16 25 48](https://github.com/user-attachments/assets/b935c118-9cc5-493b-92d7-fa6b3ec1c186)
![Screenshot 2024-10-03 at 16 26 28](https://github.com/user-attachments/assets/70157b31-361a-4f62-b540-964c5016e6c6)

Reply:
Fair enough I guess we can keep this check then.